presto-main/src/main/java/com/facebook/presto/operator/aggregation/arrayagg/SetAggregationFunction.java

```diff
@@ -83,6 +83,12 @@ public InternalAggregationFunction specialize(BoundVariables boundVariables, int
         return generateAggregation(type, outputType);
     }
 
+    @Override
+    public boolean isCalledOnNullInput()
+    {
+        return true;
```

Comment:
We might have the same issue with SET_UNION as well. I mean it maybe missing this override.

Reply:
Looks like set_union ignores NULL arrays:
presto:deltoid3> SELECT set_union(e) FROM (VALUES ARRAY[1], CAST(NULL AS ARRAY<BIGINT>)) AS t(e);
 _col0
-------
 [1]
(1 row)

It does preserve NULL array element. But in this case, its input is the array which is not NULL. 
So looks like returning false for isCalledOnNullInput() is correct for set_union?

Reply:
It has the same issue as set_agg:
```
presto:di> select set_union(if(y > 1, x)) from (select array[1] x, 1 y union all select null, 1 y);
 _col0
-------
 []
(1 row)

Query 20210805_194306_40996_sptmi, FINISHED, 1 node
Splits: 15 total, 15 done (100.00%)
0:00 [0 rows, 0B] [0 rows/s, 0B/s]

presto:di> select set_union(x) filter (where y > 1) from (select array[1] x, 1 y union all select null, 1 y);
 _col0
-------
 NULL
(1 row)
```

Reply:
Added override function. Thanks for catching this and figuring out the repro example.